lib/ui/page/buy.dart

```diff
@@ -0,0 +1,437 @@
+          },
+        ),
+      );
+    }
+
+    Future<WyreQuote> queryOrderQuote(bool byFiat) async {
+      final data = {
+        'sourceCurrency': fiat.value.name,
+        'destCurrency': asset.value.symbol,
+        'dest': 'ethereum:${asset.value.destination}',
+        'country': country,
+        'accountId': Env.wyreAccount,
+        'walletType': type.value.forQuote(),
+        'amountIncludeFees': 'true',
+      };
+      if (byFiat) {
+        data['sourceAmount'] = fiatController.text;
+      } else {
+        data['destAmount'] = cryptoController.text;
+      }
+      final quote =
+          await WyreClient.instance.api.getOrderReservationQuote(data);
+      return quote;
+    }
+
+    useEffect(() {
+      void updateAmount() {
+        _debouncer.run(() async {
+          final amount = double.tryParse(fiatController.text) ?? 0;
+          if (amount == 0) {
+            cryptoController.text = '';
+            wyreQuote.value = null;
+            return;
+          }
+
+          if (fiatFocusNode.hasFocus) {
+            final quote = await queryOrderQuote(true);
+            cryptoController.text = quote.destAmount.toString();
+            wyreQuote.value = quote;
+            lastQuoteByFiat.value = true;
+          }
+        });
+      }
+
+      fiatController.addListener(updateAmount);
+      return () {
+        fiatController.removeListener(updateAmount);
+      };
+    }, [fiatController]);
+
+    useEffect(() {
+      void updateAmount() {
+        _debouncer.run(() async {
+          final amount = double.tryParse(cryptoController.text) ?? 0;
+          if (amount == 0) {
+            fiatController.text = '';
+            wyreQuote.value = null;
+            return;
+          }
+
+          if (cryptoFocusNode.hasFocus) {
+            final quote = await queryOrderQuote(false);
+            fiatController.text = quote.sourceAmount.toString();
+            wyreQuote.value = quote;
+            lastQuoteByFiat.value = false;
+          }
+        });
+      }
+
+      cryptoController.addListener(updateAmount);
+      return () {
+        cryptoController.removeListener(updateAmount);
+      };
+    }, [cryptoController]);
+
+    return Scaffold(
+        backgroundColor: context.colorScheme.background,
+        appBar: MixinAppBar(
+          backgroundColor: context.colorScheme.background,
+          title: Text(
+            context.l10n.buy,
+            style: TextStyle(
+              color: context.colorScheme.primaryText,
+              fontSize: 18,
+              fontWeight: FontWeight.w600,
+            ),
+          ),
+          leading: const MixinBackButton2(),
+        ),
+        body: Container(
+          padding: const EdgeInsets.symmetric(vertical: 0, horizontal: 20),
+          decoration: BoxDecoration(
+            borderRadius:
+                const BorderRadius.vertical(top: Radius.circular(topRadius)),
+            color: context.colorScheme.background,
+          ),
```

Comment:
the Scaffold and Container seems could move out of this widget. Maybe this can reduce the nesting level of part of this widget

Reply:
separate `body` from other stuffs